src/libraries/System.Private.CoreLib/src/System/RuntimeType.cs

```diff
@@ -731,7 +731,7 @@ public override bool IsAssignableFrom([NotNullWhen(true)] Type? c)
 
         private static void ThrowIfTypeNeverValidGenericArgument(RuntimeType type)
         {
-            if (type.IsPointer || type.IsByRef || type == typeof(void))
+            if (type.IsPointer || type.IsFunctionPointer || type.IsByRef || type == typeof(void))
```

Comment:
I seem to remember it being discussed, but why doesn't IsPointer return true for a function pointer? That seems unexpected.

Reply:
It would make it complicated to test for ordinary pointers (would need `IsPointer && !IsFunctionPointer`). People who test `IsPointer` might expect to find an element type. Fundamentally byrefs are also pointers and they don't report true either.

ECMA spec views them like this:

![image](https://github.com/dotnet/runtime/assets/13110571/e7c9bff0-e335-494b-b053-85afa6b48c27)

Reply:
> People who test IsPointer might expect to find an element type.

They might. And people who test IsPointer on a function _pointer_ might expect it to return true. Either way there _could_ be broken expectations.

> Fundamentally byrefs are also pointers and they don't report true either.

But we don't call them pointers in the surface area. That's the sticking point. If there's am IsFoo, and also an IsMoreSpecificFoo, purely from a naming perspective it's very confusing if the former returns false when the latter returns true.

Reply:
There was a long discussion about this in https://github.com/dotnet/runtime/issues/69273 and also during API review. There is no perfect solution for this.

Reply:
There isn't, but I think where we landed isn't great. Thankfully it won't impact most people.